src/internal/storage/fileset/storage.go

```diff
@@ -185,39 +181,56 @@ func (s *Storage) CloneTx(tx *pachsql.Tx, id ID, ttl time.Duration) (*ID, error)
 	}
 }
 
-// Flatten takes a list of IDs and replaces references to composite FileSets
-// with references to all their layers inplace.
-// The returned IDs will only contain ids of Primitive FileSets
-func (s *Storage) Flatten(ctx context.Context, ids []ID) ([]ID, error) {
+// FlattenAll is like Flatten, but collects the primitives to return to the user.
```

Comment:
nit, could we move this below the `Flatten` definition. It just reads better imo.